### PR TITLE
Remove unused classifiers from filter list

### LINKED ISSUE #1386: Remove unused classifiers from filter list
We currently show all trove classifiers in the search filter panel, despite the fact that some are not applied to any projects in the DB.

It would be better to only show those classifiers that are actually applied to projects, so we avoid filtering by a classifier and returning an empty result.

### PR DESCRIPTION
**Pull request by berkerpeksag**
Fixes #1386

### CODE REVIEW

**Comment by dstufft** [comment]
Other than ``make lint`` failing, this looks good to me.

File: tests/unit/test_views.py [hunk 1]
```diff
@@ -293,6 +293,13 @@ def test_with_classifiers(self, monkeypatch, db_request, page):
         classifier2 = ClassifierFactory.create(classifier="foo :: baz")
         classifier3 = ClassifierFactory.create(classifier="fiz :: buz")
 
+        project = ProjectFactory.create()
+        release1 = ReleaseFactory.create(project=project)
+        release1.created = datetime.date(2011, 1, 1)
+        release1._classifiers.append(classifier1)
+        release1._classifiers.append(classifier2)
+        release1._classifiers.append(classifier3)
```
Comment (berkerpeksag) [inline tests/unit/test_views.py]: Should I left `classifier3` alone and add an assert to test that it's not in `search(db_request)['available_filters']`? Right now, we are not actually testing the new behavior.

Comment (dstufft) [inline tests/unit/test_views.py]: Yea that sounds good.

**Comment by berkerpeksag** [comment]
Fixed the style issues and tweaked the test case. Thanks.

**Pull request merged**